packages/cozy-stack-client/src/SharingCollection.js

```diff
@@ -0,0 +1,99 @@
+import DocumentCollection, { normalizeDoc } from './DocumentCollection'
+import { uri } from './utils'
+
+const normalizeSharing = sharing => normalizeDoc(sharing, 'io.cozy.sharings')
+
+export default class SharingCollection extends DocumentCollection {
+  async findByDoctype(doctype) {
+    const resp = await this.client.fetch(
+      'GET',
+      uri`/sharings/doctype/${doctype}`
+    )
+    return {
+      ...resp,
+      data: resp.data.map(s => normalizeSharing(s))
+    }
+  }
+
+  async share(document, recipients, sharingType, description) {
+    const resp = await this.client.fetch('POST', '/sharings/', {
+      data: {
+        type: 'io.cozy.sharings',
+        attributes: {
+          description,
+          rules: getRulesFor(document, sharingType)
+        },
+        relationships: {
+          recipients: {
+            data: recipients.map(({ id, _type }) => ({ id, type: _type }))
+          }
+        }
+      }
+    })
+    return { data: normalizeSharing(resp.data) }
+  }
+
+  revokeRecipient(sharing, recipientEmail) {
+    const memberIndex = sharing.attributes.members.findIndex(
+      m => m.email === recipientEmail
+    )
+    return this.client.fetch(
+      'DELETE',
+      uri`/sharings/${sharing._id}/recipients/${memberIndex}`
+    )
+  }
+
+  revokeSelf(sharing) {
+    return this.client.fetch(
+      'DELETE',
+      uri`/sharings/${sharing._id}/recipients/self`
+    )
+  }
+}
+
+const isFile = ({ _type, type }) =>
+  _type === 'io.cozy.files' || type === 'directory' || type === 'file'
+
+const isDirectory = ({ type }) => type === 'directory'
+
+const getBehaviorsFor = (document, sharingType) => {
```

Comment:
What does _behavior_ mean in the context of a sharing?

Reply:
Well, with the previous sharing API, you had to define a bunch of permissions to characterize the sharing (RO/RW). Now, we have to define `rules` with properties that sets the "permissions", something like `{ add: 'sync', update: 'sync', remove: 'sync' }`. I called these properties a **behavior**, but I'd like to find a better name. Any idea?

Reply:
Let's do a little brainstorming here:

the configuration object for the sharing describes some __strategies__: synchronize, revoke, etc. At least that's an __aspect__ of the sharing.

So I vote for `getSharginAspect` or `getStrategies`. But I do not have much hindsight for sharing and its lexical field. Maybe `getBehaviors` is strong enough to express what this function does.

Reply:
The stack doc talks about behavior here: https://docs.cozy.io/cozy-stack/sharing-design.html, but I guess they have the same lexical problem. Aspect looks too generic to me, strategy is better but not perfect.

Reply:
Policy ?

Reply:
I found this confusing too, may I suggest adding a comment pointing to the doc mentioned above, or briefly describing what a *behavior* is here?

Reply:
policy is great, thanks @ptbrowne!